assets/js/base/components/text-input/validated-text-input.tsx

```diff
@@ -39,8 +62,8 @@ const ValidatedTextInput = ( {
 		getValidationErrorId,
 	} = useValidationContext();
 
-	const textInputId = id || 'textinput-' + instanceId;
-	errorId = errorId || textInputId;
+	const textInputId = id !== undefined ? id : 'textinput-' + instanceId;
+	const errorIdString = errorId !== undefined ? errorId : textInputId;
```

Comment:
This change was made because TS still couldn't properly infer that after this line, `errorId` was a string. It still thought it would be `string | undefined`.